File: src/controllers/player_controller.js
define(function() { return function PlayerController(player) {
		this.player = player;
		var that = this;

		this.keyEvent = function(key) {
			key = key || window.event;
			var keyCode = key.keyCode;
			switch(keyCode) {
				case 37: // left
					that.player.move("l");
					break;
				case 39: // right
					that.player.move("r");
					break;
				case 38: // up
					that.player.move("u");
					break;
				case 40: // down
					that.player.move("d");
					break;
				default:
					//console.log(that.player);
					//alert("caught unexpected keycode "+keyCode)
			}
		}

	}
});

File: src/controllers/player_controller.dart
library playerController;

import "../models/player.dart";
import "dart:html";

class PlayerController {
	Player p;

	PlayerController(this.p);

	keyEvent(int keyCode, bool upOrDown) {
		Action action;
		switch (keyCode) {
			case KeyCode.LEFT:
				action = Action.LEFT;
				break;
			case KeyCode.RIGHT:
				action = Action.RIGHT;
				break;
			case KeyCode.SPACE:
				action = Action.JUMP;
				break;
			case KeyCode.D:
				action = Action.SHOOT_RIGHT;
				break;
			case KeyCode.A:
				action = Action.SHOOT_LEFT;
				break;
		}

		if (action == null) return;

		if (upOrDown) p.actions.remove(action);
		else p.actions.add(action);
	}
}
